monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -81,6 +88,19 @@ def __init__(
         self._session = requests.Session()
         retry_config = Retry(retries)
         self._session.mount("https://", HTTPAdapter(max_retries=retry_config))
+        self._add_request_functions()
+
+    def _add_request_functions(self):
+        for request_method in RequestMethod:
+            request_method_name = str.lower(request_method.name)
+
+            def __get_request_sending_function(method: RequestMethod) -> Callable:
+                def inner(*args, **kwargs):
+                    return self._send_request(method, *args, **kwargs)
+
+                return inner
+
+            setattr(self, f"_{request_method_name}", __get_request_sending_function(request_method))
 
     @handle_island_errors
     def connect(
```

Comment:
Why not use the same logic for this GET?

Reply:
We probably could and should, but we'll have to set `self._api_url = f"https://{island_server}/api"` first. 
This might leave the client in a state where `self._api_url` is set, but a connection couldn't be made. It's a little downside. Ideally we shouldn't initialize object properties outside of the constructor, but I don't think there's a substantial difference in this case